src/Data/Text.hs

```diff
@@ -1623,7 +1629,9 @@ group = groupBy (==)
 -- | /O(n)/ Return all initial segments of the given 'Text', shortest
 -- first.
 inits :: Text -> [Text]
-inits t@(Text arr off len) = loop 0
+inits t@(Text arr off len) 
+    | null t = [empty]
+    | otherwise = loop 0
     where loop i | i >= len = [t]
                  | otherwise = Text arr off i : loop (i + iter_ t i)
```

Comment:
this too?

Reply:
Could you expand on what you mean here? The empty check is already outside the loop afaict.

Btw I'm trying to prefer `null t` over `len == 0` as I think it can make rules for whereas the fields are opaque, but I can go another way

Reply:
I mean this starts with `loop 0` which constructs `Text arr off 0`. The head of the list is always `empty`, so `inits` could even be lazy until the next element is forced.

Reply:
Ah I see. Great spot!

Reply:
Done in 9e3622036a2a157fa49dbb68bcb5e377bccd8f1a